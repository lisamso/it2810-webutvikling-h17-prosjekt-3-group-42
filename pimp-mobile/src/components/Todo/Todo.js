import React, { Component } from 'react';
import { Text, View } from "react-native";
import PropTypes from 'prop-types';
import { styles } from '../../styles.js';

class Todo extends Component {

  render() {

    return(
      <View style={styles.todo}>
        <Text style={[styles.text, styles.turquoise]}>{this.props.text}</Text>
      </View>
    )
  }
}

Todo.propTypes = {
  text: PropTypes.string,
}

export default Todo