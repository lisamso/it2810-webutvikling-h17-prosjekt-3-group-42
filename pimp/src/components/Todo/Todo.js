import React from 'react';
import PropTypes from 'prop-types';

export const Todo = ({todo, remove}) => {
  return (
    <div className='box-item'>
      {todo.value}
      <p></p>
      <button className='button-add-new-todo'
        onClick={()=> {
          remove(todo.id)
        }}>
				Task done
      </button>
    </div>
  );
};

export const TodoList = ({todos, remove}) => {
  let allTodos = [];
  if(todos.length > 0) {
    allTodos = todos.map(todo => {
      return (
        <Todo key = {todo.id}
          todo={todo}
          remove={remove}
        />);
    });
  }
  else {
    allTodos.push(<h3 align="center">No more tasks</h3>);
  }

  return (<div> {allTodos} </div>);
};

Todo.propTypes = {
  todo: PropTypes.object,
  remove: PropTypes.func
};

TodoList.propTypes = {
  todos: PropTypes.array,
  remove: PropTypes.func
};

export default Todo;
