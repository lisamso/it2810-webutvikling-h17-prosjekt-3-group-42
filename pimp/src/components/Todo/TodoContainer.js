import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Box from '../Box';

const Todo = ({todo, remove}) => {
  return (
    <div className='box-item'>
      {todo.value}
      <p></p>
      <div id = "button-todo">
        <button className='button-add-new-todo'
          onClick={()=> {
            remove(todo.id)
          }}>
        </button>
      </div>
    </div>
  );
};


const List = ({todos, remove}) => {
  let allTodos = [];
  if(todos.length > 0) {
    allTodos = todos.map(todo => {
      return ( <Todo key = {todo.id} todo={todo} remove={remove} /> );
    });
  } else {
    allTodos.push(<h3 align="center">No more tasks</h3>);
  }

  return (
    <div>
      {allTodos}
    </div>
  );
};


class TodoContainer extends Component {
  constructor(props) {
    super(props);

    const introData = [
      {
        id: -1,
        value: "Add new todos and come " +
        "back any time later, I will save them for you!"
      }
    ];


    const localStorageData = localStorage.todos
      && JSON.parse(localStorage.todos);

    this.state = {
      data: localStorageData || introData,
      value: ''
    };

    // binding metoder
    this.addNewTodo = this.addNewTodo.bind(this);
    this.removeFinishedTodo = this.removeFinishedTodo.bind(this);
    this.handleChangeForTodo = this.handleChangeForTodo.bind(this);
    this.handleAddingNewTodo = this.handleAddingNewTodo.bind(this);
  }
  handleChangeForTodo(event) {
    this.setState({
      value: event.target.value
    });
  }

  handleAddingNewTodo() {
    if(this.input.value !== '') {
      this.addNewTodo(this.input.value);
      this.setState({
        value: ''
      });
      this.input.placeholder = "Legg til todo elementer her";
    }
  }

  // Handler for å oppdatere dataen
  updateLocalData() {
    if (typeof(Storage) !== "undefined")
      localStorage.todos = JSON.stringify(this.state.data);
  }

  // Handler for å legge til ny todo
  addNewTodo(value) {
    let id;
    if (typeof(Storage) !== "undefined") {
      id = Number(localStorage.count);
      localStorage.count = Number(localStorage.count) + 1;
    } else {
      id = window.id++;
    }

    const todo = {
      value: value,
      id: id
    };

    this.state.data.push(todo);
    this.setState({
      data: this.state.data
    }, () => {
      this.updateLocalData();
    });
  }

  // Handler for å fjerne todo
  removeFinishedTodo(id) {
    const list = this.state.data.filter(todo => {
      if (todo.id !== id)
        return todo;
    });
    this.setState({
      data: list
    }, () => {
      this.updateLocalData();
    });
  }


  componentDidMount() {
    if(!localStorage.todos) {
      localStorage.todos = JSON.stringify(this.state.data);
    }
    if(!localStorage.count) {
      localStorage.count = 0;
    }

  }



  render() {
    const todoForm =
      <div className='Todo'>
        <div className = 'group'>
          <input className='input'
            type='text'
            required
            name='title'
            ref={node => {
              this.input = node;
            }}
            value={this.state.value}
            placeholder="Legg til todo elementer her "
            autoComplete="off"
            onChange={this.handleChangeForTodo}/>
          <span className='bar'></span>


        </div>
        <div className='group'>
          <List todos={this.state.data}
            remove={this.removeFinishedTodo} />
        </div>
      </div>


    const addNewTodoButton = <button
      onClick={this.handleAddingNewTodo}
    >
      Add a new Todo
    </button>

    return (
      <div className='container-item' id ='Todo'>
        <Box
          color = {this.props.borderColor}
          name = 'Your Todo'

          obj = {[todoForm]}
          button={addNewTodoButton} />
      </div>
    );
  }

}

Todo.propTypes = {
  todo: PropTypes.object,
  remove: PropTypes.func
};

List.propTypes = {
  todos: PropTypes.array,
  remove: PropTypes.func
};

TodoContainer.propTypes = {
  addNewTodo: PropTypes.string,
  borderColor: PropTypes.string
};


export default TodoContainer;

