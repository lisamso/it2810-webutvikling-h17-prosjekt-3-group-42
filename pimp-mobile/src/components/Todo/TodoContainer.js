import React, {Component} from 'react'
import { Text, TextInput, TouchableNativeFeedback, View} from "react-native";
import Modal from 'react-native-modal';
import Todo from "./Todo";
import { styles } from '../../styles.js';

class TodoContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      todos: [],
      modalVisible: false,
      text: '',
      count: 0
    };
  }

  componentDidMount() {
    this.newTodo('Press new todo to make a new todo');
  }

  toggleModal() {
    this.setState((prevState) => {
      return {modalVisible: !prevState.modalVisible};
    });
  }

  submitTodo() {
    if (this.state.text !== '') {
      this.newTodo(this.state.text);
    }
    this.setState({text: ''});
    this.toggleModal()
  }

  removeTodo(id) {
    let arr = this.state.todos;
    let index = -1;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i].props.id === id) {
        index = i;
      }
    }
    if (index !== -1) {
      arr.splice(index, 1);
    }
    this.setState({todos: arr});
  }

  newTodo(text=''){
    let todo =
      <Todo
        id={this.state.count}
        text={text}
      />;
    this.setState((prevState) => {
      return {count: prevState.count + 1}
    });
    let arr = this.state.todos;
    arr.push(todo);
    this.setState({ todos: arr })
  }

  allTodos() {
    return(
      this.state.todos.map((n,idx) =>
        <View style={styles.row_box} key={idx}>
          <View style={[styles.container, styles.turquoise_border]}>
            {n}
          </View>
          <TouchableNativeFeedback
            onPress={() => { this.removeTodo(n.props.id)}}
          >
            <View style={styles.delete_container}>
              <View style={[styles.delete_border, styles.turquoise_border]}>
                <Text style={[styles.delete_text, styles.turquoise]}>âœ“</Text>
              </View>
            </View>
          </TouchableNativeFeedback>
        </View>
      ));
  }

  render(){
    return(
      <View style={styles.box}>
        <View style={[styles.box_header, styles.turquoise_border]}>
          <Text style={[styles.title, styles.turquoise]}>Your Todos</Text>
        </View>
        <View style={[styles.box_body, styles.turquoise_border]}>
          {this.allTodos()}
        </View>
        <TouchableNativeFeedback
          onPress={() => { this.toggleModal()}}>
          <View style={[styles.box_button, styles.turquoise_background]}>
            <Text style={styles.wet_asphalt}>
              New todo
            </Text>
          </View>
        </TouchableNativeFeedback>

        <Modal
          animationIn="slideInLeft"
          animationOut="slideOutRight"
          transparent={true}
          isVisible={this.state.modalVisible}
          backDropColor="white"
          onBackdropPress={() => {this.toggleModal()}}
          onBackButtonPress={() => {this.toggleModal()}}
        >
          <View style={styles.box}>
            <View style={[styles.box_header, styles.turquoise_border]}>
              <Text style={[styles.title, styles.turquoise]}>New todo</Text>
            </View>
            <View style={[styles.box_body, styles.turquoise_border]}>
              <TextInput
                style={{height: 60, color: '#1abc9c'}}
                placeholder="Text"
                onChangeText={(text) => this.setState({text: text})}
              />
            </View>
            <TouchableNativeFeedback
              onPress={() => {this.submitTodo()}}>
              <View style={[styles.box_button, styles.turquoise_background]}>
                <Text style={styles.wet_asphalt}>
                  Confirm
                </Text>
              </View>
            </TouchableNativeFeedback>
          </View>
        </Modal>
      </View>

    )
  }
}

TodoContainer.propTypes = {

}


export default TodoContainer

