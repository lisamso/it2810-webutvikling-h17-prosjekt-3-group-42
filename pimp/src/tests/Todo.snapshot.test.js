import Todo from '../components/Todo/Todo';
import TodoContainer from '../components/Todo/TodoContainer';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

test('Todo component should render as expected', () =>{
  const remove = jest.fn();
  const todo ={value: 'Something', remove: remove};
  const component = shallow(<Todo todo={todo}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

describe('Test the functions of the Todo component.', () =>{
  it('The add-new-todoButton calls the remove function when clicked.', () =>{
    const remove = jest.fn();
    const todo ={value: 'Something', remove: remove};
    const component = shallow(<Todo todo={todo} remove={remove}/>);
    const tree = toJson(component);
    const c = component.find('.button-add-new-todo');
    c.simulate('click');
    expect(remove).toBeCalled();
  });
});
