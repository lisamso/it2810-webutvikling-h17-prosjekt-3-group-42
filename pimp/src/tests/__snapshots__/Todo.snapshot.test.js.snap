// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should render as expected 1`] = `
<div
  className="box-item"
>
  Something
  <p />
  <button
    className="button-add-new-todo"
    onClick={[Function]}
  >
    Task done
  </button>
</div>
`;
